spec/acceptance/stub_spec.rb

```diff
@@ -99,3 +99,36 @@ def using_model_without_pk
     end
   end
 end
+
+describe "a stubbed instance with association persisted in the database" do
+  include FactoryBot::Syntax::Methods
+
+  before do
+    define_model("User") do
+      has_many :posts
+    end
+    define_model("Post", user_id: :integer) do
+      belongs_to :user
+    end
+
+    FactoryBot.define do
+      factory :user
+
+      factory :post do
+        association(:user)
+      end
+    end
+  end
+
+  let(:persisted_post) { create(:post) }
+
+  subject { build_stubbed(:user, id: persisted_post.user.id) }
+
+  it "does not assign associations in the database" do
+    expect(subject.posts).to be_empty
```

Comment:
I don't think I understand this test. Why is `subject.posts` empty if the user is associated with a post?

Reply:
Ah, perhaps I misunderstood the issue. One thing I've noticed is that while objects created with `build_stubbed` can not write from the database, they can still read from the database. The goal of this PR was to separate objects created from `build_stubbed` completely from reading from the database. If this isn't actually an issue, I'll close this PR.